crates/router/src/connector/aci.rs

```diff
@@ -102,16 +107,25 @@ impl
         &self,
         req: &types::PaymentsSyncRouterData,
         _connectors: &settings::Connectors,
-    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
-        let mut header = vec![
+    ) -> CustomResult<Vec<(String, Box<dyn services::request::HeaderValue>)>, errors::ConnectorError>
+    {
+        let mut header: Vec<(_, Box<dyn services::request::HeaderValue>)> = vec![
             (
                 headers::CONTENT_TYPE.to_string(),
-                types::PaymentsSyncType::get_content_type(self).to_string(),
+                Box::new(types::PaymentsSyncType::get_content_type(self).to_string()),
             ),
-            (headers::X_ROUTER.to_string(), "test".to_string()),
+            (headers::X_ROUTER.to_string(), Box::new("test".to_string())),
         ];
-        let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
-        header.append(&mut api_key);
+        let api_key = self
+            .get_auth_header(&req.connector_auth_type)?
+            .into_iter()
+            .map(
+                |(key, value)| -> (String, Box<dyn services::request::HeaderValue>) {
+                    (key, Box::new(value))
+                },
+            );
+        // header.append(&mut api_key);
+        header.extend(api_key);
```

Comment:
I believe you'd prefer using `concat_headers()` instead.

Reply:
In this case, the `header` variable is already of type `Vec<(String, Box<dyn HeaderValue>)>` thus, directly extending it with an iterator will have the same effect as the function `concat_headers`. The function concat headers is for a case where both of the headers provided to the function are of strict types and we want to append they as well as convert them to the able mentioned type